Fails to install FindPythonLibNew.cmake when specifying DESTDIR

I'm trying to build pybind11_catkin master using catkin tools on ROS melodic/ubuntu bionic. Is that expected to fail? The workspace which get's turned into a huge deb contains MoveIt Noetic and similar so I thought it might be right to use master instead of melodic-devel?

Error is:
```
Errors << pybind11_catkin:install /home/jenkins/workspace/mikado_mikado_ros_version-2.1/Entwicklung/../catkinworkspace-tmp/log/pybind11_catkin/build.install.000.log
CMake Error at cmake_install.cmake:155 (file):
  file INSTALL cannot find
  "/home/jenkins/workspace/mikado_mikado_ros_version-2.1/catkinworkspace-tmp/devel/.private/pybind11_catkin/share/pybind11_catkin/cmake/FindPythonLibsNew.cmake".


make[2]: *** [install] Error 1
```

earlier in the logs I found
```
Linked: (/home/jenkins/workspace/mikado_mikado_ros_version-2.1/Entwicklung/../catkinworkspace-test-tmp/devel/.private/pybind11_catkin/share/pybind11_catkin/cmake/FindPythonLibsNew.cmake, /home/jenkins/workspace/mikado_mikado_ros_version-2.1/Entwicklung/../catkinworkspace-test-tmp/devel/share/pybind11_catkin/cmake/FindPythonLibsNew.cmake)
```

which is in a different workspace that we use for running the unit tests

With the workaround in the PR It successfully build without a DESTDIR/install space set and also finished with those set. Now unfortunately I get a new error:
```
Errors << moveit_core:make /home/jenkins/workspace/mikado_mikado_ros_version-2.1/Entwicklung/../catkinworkspace-tmp/log/moveit_core/build.make.000.log
In file included from /home/jenkins/workspace/mikado_mikado_ros_version-2.1/Entwicklung/extern/moveit/moveit_core/python/tools/src/pybind_rosmsg_typecasters.cpp:37:0:
/home/jenkins/workspace/mikado_mikado_ros_version-2.1/Entwicklung/extern/moveit/moveit_core/python/tools/include/moveit/python/pybind_rosmsg_typecasters.h:39:10: fatal error: pybind11/pybind11.h: No such file or directory
 #include <pybind11/pybind11.h>
          ^~~~~~~~~~~~~~~~~~~~~
compilation terminated.
make[4]: *** [python/tools/CMakeFiles/moveit_python_tools.dir/src/pybind_rosmsg_typecasters.cpp.o] Error 1
make[3]: *** [python/tools/CMakeFiles/moveit_python_tools.dir/all] Error 2
```

There is no pybind11 folder in the install space and the pybind11_catkin one is empty

The files actually ended up here:
Entwicklung/debian/mikado-ros/home/jenkins/workspace/mikado_mikado_ros_version-2.1/catkinworkspace-tmp/devel/.private/pybind11_catkin/include/pybind11_catkin/pybind11/

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,6 +12,9 @@ ExternalProject_Add(pybind11_src
              -DPYBIND11_INSTALL=ON
              -DCMAKE_INSTALL_PREFIX=${PROJECT_BINARY_DIR}/install
              -DCMAKE_INSTALL_INCLUDEDIR=${CATKIN_DEVEL_PREFIX}/include/${PROJECT_NAME}
+  # Workaround if DESTDIR is set
+  # See https://gitlab.kitware.com/cmake/cmake/-/issues/18165.
+  INSTALL_COMMAND ${CMAKE_MAKE_PROGRAM} DESTDIR= install
 )
 # We need to copy the entire content of the cmake/pybind11 folder to the correct place but with a different structure:
 ExternalProject_Add_Step(pybind11_src CopyToDevel

